pippo-less4j/src/main/java/ro/pippo/less4j/LessResourceHandler.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ro.pippo.less4j;
+
+import com.github.sommeri.less4j.LessCompiler;
+import com.github.sommeri.less4j.LessSource;
+import com.github.sommeri.less4j.core.ThreadUnsafeLessCompiler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ro.pippo.core.Application;
+import ro.pippo.core.RuntimeMode;
+import ro.pippo.core.route.StaticResourceHandler;
+
+import java.io.*;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * @author Daniel Jipa
+ */
+public class LessResourceHandler extends StaticResourceHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LessResourceHandler.class);
+
+    private final String resourceBasePath;
+
+    private ConcurrentMap<String, String> sourceMap = new ConcurrentHashMap<>();
+
+    ThreadUnsafeLessCompiler compiler = new ThreadUnsafeLessCompiler();
```

Comment:
Hmm.  `ThreadUnsafeLessCompiler` and we are allowing concurrent access in DEV mode.  Doesn't that seem odd?